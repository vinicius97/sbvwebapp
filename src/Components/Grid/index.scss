.grid{
  &--container{
    display: flex;
    flex-direction: column;
    flex-wrap: wrap;
    margin: auto;
    padding: 30px;
    max-width: 1200px;

    @media only screen and (min-width: 768px) {
      flex-direction: row;
    }

    &--item{
      position: relative;
      display: flex;
      flex-direction: column;
      align-items: center;
      justify-content: center;

      width: 100%;
      height: 300px;
      margin: 20px auto;

      text-decoration: none;
      background-color: #f3f3f3;

      @media only screen and (min-width: 768px){
        width: 23%;
        height: 175px;
        margin: 1%;
      }

      &__title{
        position: absolute;
        left: 0;
        right: 0;
        bottom: 0;
        padding: 20px;
        width: 100%;

        color: rgba(150, 150, 150, 0.68);
        font-size: 14px;
        font-weight: bold;
        text-align: center;

        z-index: 1;
      }

      &__play{
        width: 100%;
        height: 100%;

        opacity: 0.3;

        background-image: url('./images/play.svg');
        background-position: center;
        background-repeat: no-repeat;
        background-size: 60px;

        transition: all ease-in-out 0.1s;
        z-index: 2;

        &:hover{
          background-size: 70px;
          opacity: 1;
        }
      }
    }
  }
}