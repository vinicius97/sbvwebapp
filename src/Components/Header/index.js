import React from 'react'
import { Link } from 'react-router-dom'
import PropTypes from 'prop-types'

import './index.scss'
import Logotipo from  './images/sambatech.svg'

const Header = ({homePage}) => (
  <header className={`header--container`}>
    {homePage && (
      <Link className={'header--container__back'} to={homePage} >
        Home
      </Link>
    )}
    <img
      alt={`SambaTech`}
      className={`header--container__logo`}
      src={Logotipo} />
  </header>
)

Header.propTypes = {
  homePage: PropTypes.string
}

export { Header }