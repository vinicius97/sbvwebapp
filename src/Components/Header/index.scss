.header{
  &--container{
    position: relative;
    display: flex;
    height: 120px;
    background: #ffffff;
    box-shadow: 0 0 15px -5px rgb(0, 0, 0);
    z-index: 1;

    &__logo{
      margin: auto;
      font-size: 23px;
      color: #ffffff;
    }

    &__back{
      position: absolute;
      top: 0;
      bottom: 0;
      margin: auto 20px;
      height: 40px;
      background: rgba(0, 0, 0, 0.1);
      display: flex;
      align-items: center;
      justify-content: center;
      padding: 20px;
      border-radius: 20px;
      text-decoration: none;
      color: #9a9a9a;
      font-weight: bold;
    }
  }
}