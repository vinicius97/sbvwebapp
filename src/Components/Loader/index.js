import React from 'react'
import DefaultIcon from './images/default.svg'
import ComponentIcon from './images/component.svg'

export const Loader = ({type}) => (
  (type === 'component') ? (
    <img className={`loader--icon`} src={ComponentIcon} alt='Carregando' />
  ) : (
    <img className={`loader--icon`} src={DefaultIcon} alt='Carregando' />
  )
)
