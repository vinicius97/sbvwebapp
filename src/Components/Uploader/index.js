import React, { Component, Fragment } from 'react'
import { connect } from 'react-redux'

import './index.scss'

export class Uploader extends Component{

  static defaultProps = {
    upload: {
      status: ''
    }
  }

  state = {
    error: '',
    file: null,
    title: null,
    showForm: false
  }

  handleOnFormSubmit = (e) => {
    e.preventDefault()

    const { file, title } = this.state
    if(!file) {
      this.handleError('Selecione um arquivo para envio')
      return
    }
    if(!title) {
      this.handleError('Digite um título para o vídeo')
      return
    }

    this.handleError('')
    this.handleUpload(file, title)

    this.setState({
      error: '',
      file: '',
      title: '',
      showForm: false
    })
  }

  handleError = (error) => {
    this.setState({error})
  }

  handleOnChangeTitle = (e) => {
    this.state.error && this.handleError('')
    this.setState({title: e.target.value})
  }

  handleOnChangeFile = (e) => {
    this.state.error && this.handleError('')
    this.setState({file: e.target.files[0]})
  }

  handleUpload = async (file, title) => {
    this.props.uploadFile({file, title})
  }

  handleShowForm = () => {
    this.setState((s) => ({
      showForm: !s.showForm
    }))
  }

  render() {

    const { file } = this.state
    const showForm = this.state.showForm
    const formClass = showForm ? `uploader--form uploader--form--show` : `uploader--form`
    const maskClass = showForm ? `uploader--form--mask` : `uploader--form--mask__hide`
    const inputFileClass = file ?
      (`uploader--form--input--file uploader--form--input--file--filled`)
      :
      (`uploader--form--input--file uploader--form--input--file--empty`)

    return (
      <Fragment>
        <button
          className={`uploader--form__button`}
          onClick={this.handleShowForm}
          type='submit'>
          Enviar vídeo
        </button>

        <div className={maskClass} onClick={this.handleShowForm} />

        {showForm && (
          <form className={formClass} onSubmit={this.handleOnFormSubmit}>
            <div className={`uploader--form__error`}>
              {this.state.error}
            </div>

            <input
              type='text'
              name='title'
              className={`uploader--form--input__text`}
              onKeyUp={this.handleOnChangeTitle}
              placeholder='Título do vídeo'/>

            <input
              className={inputFileClass}
              type='file'
              name='video'
              onChange={this.handleOnChangeFile} />

            <button
              className={`uploader--form__button`}
              type='submit'>
              Upload
            </button>
          </form>
        )}
      </Fragment>
    )
  }
}

const mapState = state => ({})

const mapDispatch = ({ upload: { uploadFile }}) => ({
  uploadFile
})

export default connect(mapState, mapDispatch)(Uploader)