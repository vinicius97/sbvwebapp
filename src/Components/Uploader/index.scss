.uploader--form{

  display: none;

  &--show{
    display: flex;
    flex-direction: column;
    margin: auto;
    padding: 30px;
    max-width: 360px;
    height: 330px;

    background: #fff;
    box-shadow: 0px 0px 30px 0px rgba(0, 0, 0, 0.4);
    border-radius: 5px;

    position: absolute;
    top: 0;
    bottom: 0;
    right: 0;
    left: 0;
    z-index: 4;
  }

  &--mask{
    background: rgba(102, 60, 171, 0.7);
    position: absolute;
    left: 0;
    right: 0;
    top: 0;
    bottom: 0;
    margin: auto;
    z-index: 3;
  }

  &__error{
    position: absolute;
    top: -50px;
    left: 0;
    width: 100%;
    padding: 10px;
    margin: 10px auto;

    background: rgb(255, 107, 107);
    border: 1px solid red;
    border-radius: 5px;

    color: #ffffff;
    font-weight: bold;
    font-size: 12px;
    text-align: center;

    &:empty{
      display: none;
    }
  }

  &__status{
    position: absolute;
    top: -50px;
    left: 0;
    width: 100%;
    padding: 10px;
    margin: 10px auto;

    background: rgba(255, 0, 0, 0.15);
    border: 1px solid red;
    border-radius: 5px;

    color: red;
    font-weight: bold;
    font-size: 12px;
    text-align: center;

    &:empty{
      display: none;
    }
  }

  &--input{
    &__text{
      background: rgba(177, 177, 177, 0.1);
      border: none;

      width: 100%;
      margin: 10px auto;
      padding: 10px;
      outline: none;
    }
    &--file{
      position: relative;
      width: 100%;
      height: 200px;

      background: rgba(177, 177, 177, 0.1);
      border: none;

      margin: 10px auto;
      padding: 10px;

      @mixin before($message) {
        content: $message;
        position: absolute;
        left: 0;
        right: 0;
        top: 40px;
        bottom: 0;

        background: #f7f7f7;
        color: #a0a0a0;
        font-weight: bold;
        font-size: 14px;
        text-align: center;

        padding: 20px;

        display: flex;
        align-items: center;
        justify-content: center;

        cursor: pointer;
      }

      &--filled{
        &:before{
          @include before('Arraste para trocar ou clique aqui');
        }
      }
      &--empty{
        &:before{
          @include before('Arraste um arquivo para c√° ou clique aqui');
        }
      }
    }
  }
  &__button{
    background: transparent;
    color: #663cab;
    font-size: 20px;

    padding: 0px 18px;
    height: 40px;
    margin: auto;

    border: 2px solid #663cab;
    border-radius: 20px;
    outline: none;

    transition: all ease-in-out 0.1s;

    &:hover{
      background: #7e53c3;
      color: #ffffff;
      cursor: pointer;

      transition: all ease-in-out 0.1s;
    }
  }
}