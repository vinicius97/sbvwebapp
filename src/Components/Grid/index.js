import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import Uploader from '../Uploader'
import { Loader } from '../Loader'
import './index.scss'

const Grid = ({list, uploadList}) => (
  <div className={`grid--container`}>
    <Uploader />

    {uploadList.map((item, index) => {
      if(item.status.toLowerCase() !== 'finalizado'){
        return (
          <div className={`grid--container--item grid--container--item--processing`} key={index}>
            <div className={`grid--container--item__title`}>
              {item.title}
            </div>
            <Loader />
            <div className={`grid--container--item__loading`}>
              {item.status}
            </div>
          </div>
        )
      }else{
        return(
          <Link to={`/video/${item._id}`} className={`grid--container--item`} key={index}>
            <div className={`grid--container--item__title`}>
              {item.title}
            </div>
            <div className={`grid--container--item__play`} />
          </Link>
        )
      }
    })}

    {list.map((item, index) => (
      <Link to={`/video/${item._id}`} className={`grid--container--item`} key={index}>
        <div className={`grid--container--item__title`}>
          {item.title}
        </div>
        <div className={`grid--container--item__play`} />
      </Link>
    ))}
  </div>
)

Grid.propTypes = {
  list: PropTypes.array,
  uploadList: PropTypes.array
}

Grid.defaultProps = {
  uploadList: []
}

export { Grid }