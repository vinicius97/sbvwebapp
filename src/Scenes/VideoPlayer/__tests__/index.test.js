import React from 'react'
import Adapter from 'enzyme-adapter-react-16'
import { configure, shallow } from 'enzyme'

import { VideoPlayer } from '../index'
import { Header } from '../../../Components/Header'

configure({adapter: new Adapter()})

describe('<VideoPlayer />', () => {

  const wrapper = shallow(<VideoPlayer player={{video: 'video.mp4'}} match={{params:{id: ''}}} loadVideo={()=>{}} />)

  it('Comparar snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('Validar carregamento de header', ()=> {
    const header = wrapper.find(Header)

    expect(header).toHaveLength(1)
    expect(header.prop('homePage')).toEqual('/')
  })

  it('Validar player HTML5', () => {
    const player = wrapper.find('.video-player--container__player')

    expect(player).toHaveLength(1)
    expect(player.prop('src')).toEqual('video.mp4')
  })
})