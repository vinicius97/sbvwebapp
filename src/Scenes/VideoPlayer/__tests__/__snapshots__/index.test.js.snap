// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VideoPlayer /> Comparar snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VideoPlayer
    loadVideo={[Function]}
    match={
        Object {
            "params": Object {
              "id": "",
            },
          }
    }
    player={
        Object {
            "video": "video.mp4",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header
          homePage="/"
/>,
        <div
          className="video-player--container"
>
          <video
                    className="video-player--container__player"
                    controls={true}
                    src="video.mp4"
          >
                    Seu navegador não suporta o elemento 
                    <code>
                              video
                    </code>
                    .
          </video>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "homePage": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <video
            className="video-player--container__player"
            controls={true}
            src="video.mp4"
>
            Seu navegador não suporta o elemento 
            <code>
                        video
            </code>
            .
</video>,
          "className": "video-player--container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Seu navegador não suporta o elemento ",
              <code>
                video
</code>,
              ".",
            ],
            "className": "video-player--container__player",
            "controls": true,
            "src": "video.mp4",
          },
          "ref": null,
          "rendered": Array [
            "Seu navegador não suporta o elemento ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "video",
              },
              "ref": null,
              "rendered": "video",
              "type": "code",
            },
            ".",
          ],
          "type": "video",
        },
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header
            homePage="/"
/>,
          <div
            className="video-player--container"
>
            <video
                        className="video-player--container__player"
                        controls={true}
                        src="video.mp4"
            >
                        Seu navegador não suporta o elemento 
                        <code>
                                    video
                        </code>
                        .
            </video>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "homePage": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <video
              className="video-player--container__player"
              controls={true}
              src="video.mp4"
>
              Seu navegador não suporta o elemento 
              <code>
                            video
              </code>
              .
</video>,
            "className": "video-player--container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Seu navegador não suporta o elemento ",
                <code>
                  video
</code>,
                ".",
              ],
              "className": "video-player--container__player",
              "controls": true,
              "src": "video.mp4",
            },
            "ref": null,
            "rendered": Array [
              "Seu navegador não suporta o elemento ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "video",
                },
                "ref": null,
                "rendered": "video",
                "type": "code",
              },
              ".",
            ],
            "type": "video",
          },
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
